### DISCUSSION TITLE
Launch a Cube!

### DISCUSSION THREAD

**Discussion by danigonzalem** [MicroStrategy/rest-api-playground#9] [Q&A]
Hello everybody, 
I hope someone answer me, because I never get the complete solution. I want to launch a cube using the API and when the cube is executed get the status in order to relaunch if it has been an error or notify to the users when apply.

I used this worflow:
1. Get the datetime of the last update
2. Launch the cube
3. When the cube has the status 64 (finished)
   - if updated datetime<>initial datetime then OK
   - if updated datetime=initial datetime then KO

I Want to know if this is the correct way to ensure that a cube is launched correctly, thanks!

**Comment by xdumstr** [comment]
You can use [monitor cache API](https://microstrategy.github.io/rest-api-docs/common-workflows/monitor-and-manage-cube-caches) to check the cube cache status.

**Reply by danigonzalem** [reply]
Thank you xdumstr,
we do this for checking the lastupdatetime of the cube's cache
GET $url_base+"/MicroStrategyLibrary/api/monitors/caches/cubes?clusterNode="+$clusterNameNode+"&projectIds="+$projectId 
with this we get a JSON where there are all cubes and their caches, but not always is updated this field. We check the JSON for 15 minutes  after that the job returns a KO.
Is it possible? should we do the call  "GET /api/monitors/caches/cubes/{cacheId}" better??

Bye

**Reply by elinmstr** [reply]
@danigonzalem The `lastUpdateTime` field is the last time the cube was compared to the data warehouse. To have this updated through rest api, you must execute/publish the cube to library each time [POST /api/v2/cubes/{cubeId}](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#Cubes/publishCube_2). 
In summary, this workflow

1. [GET /api/monitors/caches/cubes/{id}](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#Monitors/getSingleCubeProperties)
2. [POST /api/v2/cubes/{cubeId}](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#Cubes/publishCube_2).
3. [GET /api/monitors/caches/cubes/{id}](https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#Monitors/getSingleCubeProperties)

should work. In step 2, a new job is created each time which will update the cache after it's done.

From step 1 to step 3. The json field `lastUpdateJob` would be updated along with `lastUpdateTime ` for instance if the `lastUpdateJob` in step 1 was `100` in step 3 it would be `101` assuming no other jobs were running.

Note: Make sure the cube's state is correct. For instance, is it `active` and `loaded` in step 1 and step 3(loaded into memory).

**Reply by danigonzalem** [reply]
Hello and thanks!
My objective is avoid to set the node for looking up the caches, that is needed in /api/monitors/caches/cubes and in the /api/monitors/caches/cubes/ is needed the id of the cache of the cube, not the definition of the cube. An the id cache of the cube is in /api/monitors/caches/cubes too.

So finally we wil use this flow:
1.GET /api/cubes/{id}  -> get lastupdatetime
2.POST /api/v2/cubes/{cubeId} -> launch cube
3.HEAD /api/cubes/{cubeId}  -> verify the status 
4.GET /api/cubes/{id}  get lastupdatetime to confirm the new update date

Thanks for helping me
